src/components/ha-markdown.html

```diff
@@ -1,65 +1,59 @@
-<link rel='import' href='../../bower_components/polymer/polymer-element.html'>
-
 <link rel='import' href='../util/hass-mixins.html'>
 
 <script>
-class HaMarkdown extends window.hassMixins.EventsMixin(Polymer.Element) {
+class HaMarkdown extends window.hassMixins.EventsMixin(HTMLElement) {
   static get is() { return 'ha-markdown'; }
 
-  static get properties() {
-    return {
-      content: {
-        type: String,
-      },
+  connectedCallback() {
+    // 0 = not loaded, 1 = success, 2 = error
+    this._scriptLoaded = 0;
+    this._renderScheduled = false;
+    this._resize = () => this.fire('iron-resize');
 
-      scriptLoaded: {
-        type: Number,
-        // 0 = not loaded, 1 = success, 2 = error
-        value: 0,
-      },
-    };
+    Polymer.importHref(
+      '/static/pagedown-js.html',
+      () => { this._scriptLoaded = 1; this._render(); },
+      () => { this._scriptLoaded = 2; this._render(); },
+    );
   }
 
-  static get observers() {
-    return [
-      'updateContent(content, scriptLoaded)',
-    ];
+  set content(value) {
+    this._content = value;
+    this._render();
   }
 
-  updateContent(content, scriptLoaded) {
-    if (scriptLoaded === 1) {
-      const converter = window.Markdown.getSanitizingConverter();
-      this.innerHTML = converter.makeHtml(content);
+  _render() {
+    if (this._scriptLoaded === 0 || this._renderScheduled) return;
 
-      const walker = document.createTreeWalker(this, 1 /* SHOW_ELEMENT */, null, false);
+    this._renderScheduled = true;
 
-      while (walker.nextNode()) {
-        const node = walker.currentNode;
+    // debounce it to next microtask.
+    Promise.resolve().then(() => {
```

Comment:
Use setTimeout(, 0) or requestAnimationFrame instead? I think this is a heavy operation.

Reply:
`setTimeout` will trigger the render at the end of the event loop. `Promise.then` is executed as a microtask. [This](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) is a good blog post about the topic.

Reply:
In the future we would just do `await 0` 👍